app/src/main/java/org/oppia/android/app/topic/TopicTab.kt

```diff
@@ -6,58 +6,104 @@ import org.oppia.android.app.ui.R
+      TabConfig.ThreeTabsWithPractice to 1,
+      TabConfig.FourTabs to 2
+    )
   ),
   STUDY(
-    positionWithTwoTabs = 1,
-    positionWithFourTabs = 3,
     tabLabelResId = R.string.study,
     tabIconResId = R.drawable.ic_revision_icon_24dp,
-    contentDescriptionResId = R.string.revision_tab_content_description
+    contentDescriptionResId = R.string.revision_tab_content_description,
+    positions = mapOf(
+      TabConfig.TwoTabs to 1,
+      TabConfig.ThreeTabsWithInfo to 2,
+      TabConfig.ThreeTabsWithPractice to 2,
+      TabConfig.FourTabs to 3
+    )
   );
 
+  /** Enum representing different tab configurations. */
+  enum class TabConfig {
+    TwoTabs,
+    ThreeTabsWithInfo,
+    ThreeTabsWithPractice,
+    FourTabs;
+
+    companion object {
+      fun getConfig(enableInfo: Boolean, enablePractice: Boolean) = when {
+        enableInfo && enablePractice -> FourTabs
+        enableInfo -> ThreeTabsWithInfo
+        enablePractice -> ThreeTabsWithPractice
+        else -> TwoTabs
+      }
+    }
+  }
+
+  /** Returns the tab position based on enabled tabs. */
+  fun getPosition(enableTopicInfoTab: Boolean, enableTopicPracticeTab: Boolean): Int {
+    val config = TabConfig.getConfig(enableTopicInfoTab, enableTopicPracticeTab)
+    return positions[config] ?: -1 // -1 indicates that the tab is not available
+  }
+
   companion object {
     /**
      * Returns the [TopicTab] corresponding to the specified tab position, considering whether the
-     * info and practice tabs are enabled per [enableExtraTopicTabsUi].
+     * info and practice tabs are enabled per [enableTopicInfoTab] and [enableTopicPracticeTab]
+     * respectively.
      */
-    fun getTabForPosition(position: Int, enableExtraTopicTabsUi: Boolean): TopicTab {
-      return checkNotNull(
-        values().find {
-          position == if (enableExtraTopicTabsUi) {
-            it.positionWithFourTabs
-          } else it.positionWithTwoTabs
-        }
-      ) { "No tab corresponding to position: $position" }
+    fun getTabForPosition(
+      position: Int,
+      enableTopicInfoTab: Boolean,
+      enableTopicPracticeTab: Boolean
+    ): TopicTab {
+      return values().find {
+        it.getPosition(enableTopicInfoTab, enableTopicPracticeTab) == position
+      } ?: error("No tab corresponding to position: $position")
     }
 
-    /** Returns the number of active tabs considering [enableExtraTopicTabsUi]. */
-    fun getTabCount(enableExtraTopicTabsUi: Boolean) =
-      if (enableExtraTopicTabsUi) values().size else values().size - 2
+    /**
+     *  Returns the number of active tabs considering [enableTopicInfoTab] and
+     *  [enableTopicPracticeTab].
+     */
+    fun getTabCount(enableTopicInfoTab: Boolean, enableTopicPracticeTab: Boolean) =
+      if (enableTopicInfoTab && enableTopicPracticeTab)
+        values().size
+      else if (enableTopicInfoTab || enableTopicPracticeTab)
+        values().size - 1
+      else values().size - 2
```

Comment:
Making use of the new structures introduced:
```suggestion
   fun getTabCount(enableTopicInfoTab: Boolean, enableTopicPracticeTab: Boolean): Int {
  val config = TabConfig.getConfig(enableTopicInfoTab, enableTopicPracticeTab)
  return values().count { it.positions[config] != -1 }
}
```

This is less brittle/more flexible.

Reply:
Updated.